### DISCUSSION TITLE
HTTPResponse with content type application/json.

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5330] [Q&A]
Please provide sample workflow json for returning HTTPResponse with content type application/json. Content needs to set values from last activity result

**Comment by glime-ai** [comment]
To create a sample workflow JSON that returns an HTTP response with the content type `application/json` and sets the content values from the last activity result, you can use the following JSON structure:

```json
{
  "name": "Sample JSON Response Workflow",
  "activities": [
    {
      "name": "SomeActivity",
      "type": "SomeActivityType",
      "properties": {
        // ... properties specific to this activity
      },
      "outputs": {
        "Done": {
          "outcome": "Done",
          "nextActivityName": "WriteJsonResponse"
        }
      }
    },
    {
      "name": "WriteJsonResponse",
      "type": "WriteHttpResponse",
      "properties": {
        "StatusCode": {
          "syntax": "Literal",
          "expression": "200"
        },
        "ContentType": {
          "syntax": "Literal",
          "expression": "application/json"
        },
        "Content": {
          "syntax": "Json",
          "expression": "${lastResult()}" // Assuming lastResult() is a function that gets the result of the last activity
        }
      }
    }
  ],
  "connections": [
    {
      "sourceActivityName": "SomeActivity",
      "targetActivityName": "WriteJsonResponse",
      "outcome": "Done"
    }
  ]
}
```

In this JSON:

- Replace `SomeActivity` and `SomeActivityType` with the actual activity you want to use before sending the HTTP response.
- The `lastResult()` function in the `Content` property of the `WriteHttpResponse` activity is a placeholder. You should replace it with the actual expression or variable that holds the result of the last activity.
- The `WriteHttpResponse` activity is configured to return a status code of 200 and set the `ContentType` to `application/json`.
- The `connections` array defines the flow from `SomeActivity` to `WriteJsonResponse` when the outcome of `SomeActivity` is "Done".

Please note that the actual function or method to retrieve the last activity result depends on the specifics of the workflow system you are using. You may need to refer to the documentation of your workflow system to find the correct expression to use.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
